prow/flagutil/core_config.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// CoreConfigOptions holds options for interacting with Configs.
+type CoreConfigOptions struct {
+	ConfigPath string
+}
+
+// AddFlags injects config options into the given FlagSet.
+func (o *CoreConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.ConfigPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+}
+
+// Validate validates config options.
+func (o *CoreConfigOptions) Validate() error {
+	return nil
+}
+
+// Agent returns a started config agent.
+func (o *CoreConfigOptions) Agent() (agent *config.Agent, err error) {
+	agent = &config.Agent{}
+	config, err := config.Load(o.ConfigPath, "")
+	if err != nil {
+		return nil, err
+	}
+
+	err = agent.Start(o.ConfigPath, "")
```

Comment:
This whole function seems more appropriate as a function in the config package:

```golang
package config
func NewAgent(corePath, jobPath string) (*config.Agent, error) {
  cfg, err := config.Load(corepath, jobPath)
  var agent config.Agent
  err = agent.Start(corePath, jobPath)
  return &agent, nil
}

Reply:
Yeah, maybe _this_ is the DRY we really need? Flag _names_ being identical is nice, maybe we could have constants for those .. the rest .. yeah it's hard to DRY since everyone does it differently